lib/ruby_lsp/ruby_lsp_rails/code_lens.rb

```diff
@@ -102,6 +132,48 @@ def on_class_node_leave(node)
 
       private
 
+      sig { params(node: Prism::ClassNode).void }
+      def add_route_code_lenses_to_actions(node)
+        public_method_nodes = T.must(node.body).child_nodes.take_while do |node|
```

Comment:
(not a blocker) I wonder if we should use the index to get the methods names 🤔

Reply:
Something like this: https://github.com/Shopify/ruby-lsp-rails/commit/debe164fcab9201adaf5467145c71534d12ba791

Reply:
Another option may be to do it server-side:

```
irb(main):001> UsersController.action_methods
=> #<Set: {"index", "archive", "create"}>
```

Reply:
A downside of the indexing approach is that currently we only reindex on save, so we wouldn't get the the immediate feedback when editing an action definition.

Reply:
- If a developer nests a class inside a controller class, does the leave event trigger at the right time so we don't decorate the wrong classes' methods?
- Why does the ivar need to be an array? It is either in a controller or it isn't, so we could use a single boolean right?

Reply:
Yes, I think we need to make this type of information available to all listeners. It's common for them to need it.